lib/file_store/s3_store.rb

```diff
@@ -175,7 +213,17 @@ def cdn_url(url)
 
     def signed_url_for_path(path, expires_in: S3Helper::DOWNLOAD_URL_EXPIRES_AFTER_SECONDS, force_download: false)
       key = path.sub(absolute_base_url + "/", "")
-      presigned_url(key, expires_in: expires_in, force_download: force_download)
+      presigned_get_url(key, expires_in: expires_in, force_download: force_download)
+    end
+
+    def signed_url_for_temporary_upload(file_name, expires_in: S3Helper::UPLOAD_URL_EXPIRES_AFTER_SECONDS, metadata: {})
+      folder = s3_bucket_folder_path.nil? ? "" : "#{s3_bucket_folder_path}/"
+      key = "#{folder}#{upload_path}/#{TEMPORARY_UPLOAD_PREFIX}#{SecureRandom.hex}/#{file_name}"
```

Comment:
```suggestion
      key = File.join(
        folder,
        upload_path,
        TEMPORARY_UPLOAD_PREFIX,
        SecureRandom.hex,
        file_name
      )
```

Minor suggestion which I think makes the code safer and easier to read. Reduces the risk of generating an incorrect string.